### DISCUSSION TITLE
nerdctl在创建容器过程中，会产生大量 /usr/local/bin/nerdctl _NERDCTL_INTERNAL_LOGGING /var/lib/nerdctl/1935db59 进程，占用大量内存

### DISCUSSION THREAD

**Discussion by vcjia** [containerd/nerdctl#1420] [Q&A] [answered]
### Description

nerdctl在创建容器过程中，会产生大量 /usr/local/bin/nerdctl _NERDCTL_INTERNAL_LOGGING /var/lib/nerdctl/1935db59 进程，占用大量内存

### Steps to reproduce the issue

1.
2.
3.

### Describe the results you received and expected

不会产生这类进程： /usr/local/bin/nerdctl _NERDCTL_INTERNAL_LOGGING /var/lib/nerdctl/1935db59 
因为这类进程实在是太多了，占用了大量进程

### What version of nerdctl are you using?

Client:
 Version:       v0.23.0
 OS/Arch:       linux/amd64
 Git commit:    660680b7ddfde1d38a66ec1c7f08f8d89ab92c68
 buildctl:
  Version:      v0.10.4
  GitCommit:    a2ba6869363812a210fcc3ded6926757ab780b5f

Server:
 containerd:
  Version:      v1.6.8
  GitCommit:    9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:      1.1.4
  GitCommit:    v1.1.4-0-g5fd4c4d1

**Comment by Zheaoli** [comment]
Would you mind describing the problem using English?

**Comment by vcjia** [comment]
When nerdctl creates a container, a large number of /usr/local/bin/nerdctl _NERDCTL_INTERNAL_LOGGING /var/lib/nerdctl/1935db59 processes will be generated, occupying a lot of memory

**Comment by vcjia** [comment]
At the same time, a large number of useless processes such as /usr/local/bin/containerd-shim-runc-v2 -namespace default -id 1bd8240eed267bdacf0670649908da30d6f05529a1d69f5ed1d9d8f8694e0bd7 -address /run/containerd/containerd.sock.       because 1bd8240eed267bdacf0670649908da30d6f05529a1d69f5ed1d9d8f8694e0bd7 corresponding container have exited

**Comment by Zheaoli** [comment, marked as answer]
I think it's not a bug. It's a special callback to let contained push logs to the nerdctl. FYI https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/run.go#L882-L895

**Reply by vcjia** [reply]
> I think it's not a bug. It's a special callback to let contained push logs to the nerdctl. FYI https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/run.go#L882-L895

After the log is pushed, why not close the process?

**Reply by vcjia** [reply]
I found that nerdctl uses more memory than docker, if so, why should I use nerdctl?

**Reply by Zheaoli** [reply]
This process needs to exist until the container has been exited because the containerd will send the log to the process's stdin, and the nerdctl will read the log continually.
